src/hera.cpp

```diff
@@ -123,18 +124,14 @@ void execute(
   // TODO: validate for other conditions too?
   heraAssert(module.getExportOrNull(Name("main")) != nullptr, "Contract entry point (\"main\") missing.");
 
-  // Optimise
-  // PassRunner passRunner(module);
-  // passRunner.addDefaultOptimizationPasses();
-  // passRunner.run();
+  // NOTE: DO NOT use the optimiser here, it will conflict with metering
 
   // Interpet
   EthereumInterface interface(context, code, msg, result);
   ModuleInstance instance(module, &interface);
 
-  Name main = Name("main");
-  LiteralList args;
-  instance.callExport(main, args);
+  // Execute main
+  instance.callExport(Name("main"), LiteralList());
```

Comment:
This one does not build, depends on a binaryen fix. But the rest can be reviewed.